File: backend/app/modules/patients/schemas.py
import datetime as dt
import uuid
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, EmailStr


class PatientBase(BaseModel):
    nome: str
    telefone: str
    email: EmailStr | None = None
    observacoes: str | None = None


class PatientCreate(PatientBase):
    pass


class PatientUpdate(BaseModel):
    nome: str | None = None
    telefone: str | None = None
    email: EmailStr | None = None
    observacoes: str | None = None


class PatientOut(PatientBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID


class RecordAppointment(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    starts_at: dt.datetime
    ends_at: dt.datetime
    status: str
    notes: str | None


class RecordCharge(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    valor: Decimal
    descricao: str | None
    status: str
    charge_id: str
    created_at: dt.datetime


class PatientRecordOut(BaseModel):
    patient: PatientOut
    appointments: list[RecordAppointment]
    charges: list[RecordCharge]

File: apps/web/src/features/patients/schemas.ts
import { z } from "zod";

export const patientFormSchema = z.object({
  nome: z.string().min(1, "Informe o nome"),
  telefone: z.string().min(1, "Informe o telefone"),
  email: z.union([z.string().email("E-mail inválido"), z.literal("")]).optional(),
  observacoes: z.string().optional(),
});

export type PatientFormInput = z.infer<typeof patientFormSchema>;

export interface Patient {
  id: string;
  nome: string;
  telefone: string;
  email: string | null;
  observacoes: string | null;
}

export interface RecordAppointment {
  id: string;
  starts_at: string;
  ends_at: string;
  status: string;
  notes: string | null;
}

export interface RecordCharge {
  id: string;
  valor: string;
  descricao: string | null;
  status: string;
  charge_id: string;
  created_at: string;
}

export interface PatientRecord {
  patient: Patient;
  appointments: RecordAppointment[];
  charges: RecordCharge[];
}
